### PR TITLE
drivers/hd44780: move params header at the right place

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

While answering to a question on users ML about the hd44780 driver, I realized that this driver didn't have a params header file but that there's one in the tests application directory.

This PR is moving at the right place, in drivers, like it's done for others.

### Testing procedure

- Build should work at least (use one board with arduino feature):
```
$ make BOARD=arduino-zero -C tests/driver_hd44780
```
- The device should still work.

### Issues/PRs references

None

### CODE REVIEW

**Comment by aabadie** [comment]
@fjmolinas I went a little further and did some cleanup in the driver, removed the arduino feature requirement. This results in no more blacklisted board.

**Comment by fjmolinas** [comment]
> @fjmolinas I went a little further and did some cleanup in the driver, removed the arduino feature requirement. This results in no more blacklisted board.

Ok this will need to be tested though, maybe @smlng can do it?

**Comment by benpicco** [comment]
I'll give it a try!

File: drivers/hd44780/include/hd44780_params.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#define HD44780_PARAMS               {  .cols   = HD44780_PARAM_COLS,       \
+                                        .rows   = HD44780_PARAM_ROWS,       \
+                                        .rs     = HD44780_PARAM_PIN_RS,     \
+                                        .rw     = HD44780_PARAM_PIN_RW,     \
+                                        .enable = HD44780_PARAM_PIN_ENABLE, \
+                                        .data   = HD44780_PARAM_PINS_DATA }
+#endif
+
+/**
+ * @brief   LCM1602C configuration
```
Comment (aabadie) [inline drivers/hd44780/include/hd44780_params.h]: Note to myself: s/LCM1602C/HD44780/

Comment (benpicco) [reply drivers/hd44780/include/hd44780_params.h]: I think the name makes sense for the example config: LCM1602C is a concrete module with 16x2 characters whereas HD44780 is the controller chip that can be wired up with all kinds of displays.

File: drivers/hd44780/include/hd44780_params.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#define HD44780_PARAM_PIN_RW            GPIO_UNDEF
+#endif
+#ifndef HD44780_PARAM_PIN_ENABLE
+#define HD44780_PARAM_PIN_ENABLE        GPIO_PIN(PA, 9) /* Arduino D3 */
+#endif
+#ifndef HD44780_PARAM_PINS_DATA
+#define HD44780_PARAM_PINS_DATA         {   GPIO_PIN(PA, 8),  /* Arduino D4 */  \
+                                            GPIO_PIN(PA, 15), /* Arduino D5 */  \
+                                            GPIO_PIN(PA, 20), /* Arduino D6 */  \
+                                            GPIO_PIN(PA, 21), /* Arduino D7 */  \
```
Comment (benpicco) [inline drivers/hd44780/include/hd44780_params.h]: Not all platforms define a `PA` macro.
Better use `GPIO_PIN(0, …)`

**Comment by benpicco** [comment]
Still works on `mcb2300` - with that additional 1µs delay after each character.
Could you add this patch to this PR?

Also, change `PA` -> `0` to fix the build failures in CI, then this should be good to go.

**Comment by aabadie** [comment]
> Could you add this patch to this PR?

I cherry-picked the commit from #12590.
Your other comment is also addressed.

**Comment by benpicco** [comment]
Just squash already.

**Comment by aabadie** [comment]
squashed!

**Comment by benpicco** [comment]
Just tested again, works out of the box now :)

**Comment by aabadie** [comment]
@fjmolinas is on vacation and won't be back before 12th of november. His comment was addressed. So maybe just dismiss his review ?

**Comment by aabadie** [comment]
Thank you for the review @benpicco and sorry for the duplicate work.

**Comment by MrKevinWeiss** [comment]
This seems to make a hardfault on the llvm complied toolchain on the samr21-xpro with `tests/driver_hd44780/samr21-xpro:llvm`.  Results caught on [nightly](https://ci.riot-os.org/RIOT-OS/RIOT/master/ea4b78654f45a33551d32ca205f35eac0aad4f56/output/run_test/tests/driver_hd44780/samr21-xpro:llvm.txt).

We have set the label `CI: run tests`

**Comment by aabadie** [comment]
Thanks @MrKevinWeiss. I think that before this test was skipped by the CI because samr21-xpro doesn't provide the arduino feature. I'll have a look.

**Comment by aabadie** [comment]
@MrKevinWeiss, just tried `flash test` without any module on some boards and I have strange results: the hd44780 test application doesn't work on some sam0 based boards (arduino-zero, samr30-xpro) with gcc or clang. It works on nrf51 and nucleo-l073rz with both gcc and clang.
On samr21-xpro, it works with gcc but not with clang.

My guess is that something is wrong with the params structure, since it contains an array. But I'm unsure.

@benpicco, what board did you use when you tested this PR ?

**Pull request merged**